File: apiwzsk/VecWzskVExpstate.cpp
/**
	* \file VecWzskVExpstate.cpp
	* vector VecWzskVExpstate (implementation)
	* \author Catherine Johnson
	* \date created: 23 Jul 2020
	* \date modified: 23 Jul 2020
	*/

#include "VecWzskVExpstate.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 namespace VecWzskVExpstate
 ******************************************************************************/

uint VecWzskVExpstate::getIx(
			const string& sref
		) {
	string s = StrMod::lc(sref);

	if (s == "detd") return DETD;
	if (s == "maxd") return MAXD;
	if (s == "mind") return MIND;
	if (s == "regd") return REGD;

	return(0);
};

string VecWzskVExpstate::getSref(
			const uint ix
		) {
	if (ix == DETD) return("detd");
	if (ix == MAXD) return("maxd");
	if (ix == MIND) return("mind");
	if (ix == REGD) return("regd");

	return("");
};

File: japiwzsk/VecWzskVExpstate.java
/**
  * \file VecWzskVExpstate.java
  * vector VecWzskVExpstate
  * \author Catherine Johnson
  * \date created: 18 Oct 2020
  * \date modified: 18 Oct 2020
  */

package apiwzsk;

public class VecWzskVExpstate {
	public static final int DETD = 1;
	public static final int MAXD = 2;
	public static final int MIND = 3;
	public static final int REGD = 4;

	public static int getIx(
				String sref
			) {
		String s = sref.toLowerCase();

		if (s.equals("detd")) return DETD;
		if (s.equals("maxd")) return MAXD;
		if (s.equals("mind")) return MIND;
		if (s.equals("regd")) return REGD;

		return 0;
	};

	public static String getSref(
				int ix
			) {
		if (ix == DETD) return("detd");
		if (ix == MAXD) return("maxd");
		if (ix == MIND) return("mind");
		if (ix == REGD) return("regd");

		return "";
	};

};

